### PR TITLE
Sort concepts in hierarchical view by label and locale

### PR DESCRIPTION
**Pull request by UnniKohonen**
## Reasons for creating this PR

Concepts in hierarchical view are not currently sorted. This PR add functionality to sort concepts by their label according to locale.

## Link to relevant issue(s), if any

- Part of #1521 

## Description of the changes in this PR

Sort concepts in hierarchy by label and locale when they are added to hierarchy tree

Addresses requirement 6d in #1521 

## Known problems or uncertainties in this PR

Concepts are only sorted by label and locale, not notation codes or notation sorting style

## Checklist

- [ ] phpUnit tests pass locally with my changes
- [ ] I have added tests that show that the new code works, or tests are not relevant for this PR (e.g. only HTML/CSS changes)
- [ ] The PR doesn't reduce accessibility of the front-end code (e.g. tab focus, scaling to different resolutions, use of `.sr-only` class, color contrast)
- [ ] The PR doesn't introduce unintended code changes (e.g. empty lines or useless reindentation)

### CODE REVIEW

File: resource/js/tab-hierarchy.js [hunk 1]
```diff
@@ -148,6 +153,13 @@ const tabHierApp = Vue.createApp({
         height: 'calc( 100% - ' + height + 'px)',
         width: width + 'px'
       }
+    },
+    compareLabels (a, b) {
+      const labelA = a.label ? a.label : a.prefLabel
+      const labelB = b.label ? b.label : b.prefLabel
+      const lang = window.SKOSMOS.content_lang ? window.SKOSMOS.content_lang : window.SKOSMOS.lang
+
+      return labelA.localeCompare(labelB, lang)
```
Comment (osma) [inline resource/js/tab-hierarchy.js:162]: This may fail if labelA happens to be null, which currently happens for example in some of the Cypress tests. I suggest checking for the case where either labelA or labelB is null and using another value instead such as "" (empty string).

File: tests/cypress/template/vocab-home.cy.js [hunk 2]
```diff
@@ -61,7 +61,7 @@ describe('Vocabulary home page', () => {
     cy.get('#tab-alphabetical').contains('a', 'care institutions').click()
 
     // check the concept prefLabel
-    cy.get('#concept-heading h1').invoke('text').should('equal', 'care institutions')
+    cy.get('#concept-heading h1', {'timeout': 15000}).invoke('text').should('equal', 'care institutions')
```
Comment (osma) [inline tests/cypress/template/vocab-home.cy.js:64]: Is this change necessary?

**Review by osma — CHANGES_REQUESTED** [review]
null labels should be handled, otherwise this looks good!

File: tests/cypress/template/vocab-home.cy.js (hunk 2 above)
Comment (UnniKohonen) [reply tests/cypress/template/vocab-home.cy.js:64]: It was failing before without the timeout but now it isn't anymore. I'll take it out

**Review by osma — APPROVED** [review]
LGTM, please check the SonarCloud suggestions
